src/Tizen.NUI/src/public/Events/GestureDetectorManager.cs

```diff
@@ -107,34 +108,49 @@ private void init(View view)
 
     private void InternalOnTap(object sender, TapGestureDetector.DetectedEventArgs e)
     {
-      mListener.OnTap(sender, e);
+      mListener.OnTap(sender, e, mUserData);
       mTapGestureDetector.Detected -= InternalOnTap;
     }
 
     private void InternalOnLongPress(object sender, LongPressGestureDetector.DetectedEventArgs e)
     {
-      mListener.OnLongPress(sender, e);
+      mListener.OnLongPress(sender, e, mUserData);
       mLongGestureDetector.Detected -= InternalOnLongPress;
     }
 
     private void InternalOnPan(object sender, PanGestureDetector.DetectedEventArgs e)
     {
-      mListener.OnPan(sender, e);
+      mListener.OnPan(sender, e, mUserData);
       mPanGestureDetector.Detected -= InternalOnPan;
     }
 
     private void InternalOnPinch(object sender, PinchGestureDetector.DetectedEventArgs e)
     {
-      mListener.OnPinch(sender, e);
+      mListener.OnPinch(sender, e, mUserData);
       mPinchGestureDetector.Detected -= InternalOnPinch;
     }
 
     /// <summary>
     /// Gestures also work only when there is a touch event.
     /// </summary>
+    /// <param name="sender">The actor who delivered the touch event.</param>
+    /// <param name="e">The TouchEventArgs</param>
     [EditorBrowsable(EditorBrowsableState.Never)]
     public void FeedTouchEvent(object sender, View.TouchEventArgs e)
     {
+      FeedTouchEvent(sender, e, null);
+    }
+
+    /// <summary>
+    /// Gestures also work only when there is a touch event.
+    /// </summary>
+    /// <param name="sender">The actor who delivered the touch event.</param>
+    /// <param name="e">The TouchEventArgs</param>
+    /// <param name="userData">The user data object</param>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public void FeedTouchEvent(object sender, View.TouchEventArgs e, object userData)
```

Comment:
sorry but I couldn't get the purpose of this "object userData"... could you add some detail or explanation more?